File: java/cours/src/session2/td/agregation/Nat.java
package session2.td.agregation;

import session1.td.FabriqueNaturels;

import hierarchie.SemiAnneauUnitaireEuclidien;

public interface Nat extends SemiAnneauUnitaireEuclidien<Nat>, FabriqueNaturels<Nat>, EtatNaturel<Nat>{
	
	public Nat creerNatAvecEtat(EtatNaturelPur etat);
	EtatNaturelPur etat();
}

File: typescript/session2/agregation/Nat.ts
import {SemiAnneauUnitaireEuclidien} from "../../structuresAlgebriques/hierarchie";
import {FabriqueNaturels} from "../../session1/td/Nat/FabriqueNaturels";
import {EtatNaturel} from "./EtatNaturel";
import {EtatNaturelPur} from "./EtatNaturelIPur";

export interface Nat extends SemiAnneauUnitaireEuclidien<Nat>, FabriqueNaturels<Nat>, EtatNaturel<Nat>{
    creerNatAvecEtat(etat : EtatNaturelPur) : Nat;
    getEtat() : EtatNaturelPur;
}
